src/Navigation.re

```diff
@@ -14,6 +20,24 @@ external navigateWithParams: (t, string, Js.t({..})) => unit = "navigate";
 [@bs.send] external goBackToRoute: (t, string) => unit = "";
 [@bs.send] external getParam: (t, string) => Js.nullable('a) = "";
 [@bs.send] external getParamWithDefault: (t, string, 'a) => 'a = "getParam";
+[@bs.send]
+external addListener:
+  (
+    t,
+    [@bs.string] [ | `didBlur | `didFocus | `willBlur | `willFocus],
+    {
+      .
+      "action": navigationEventPayload,
+      "context": string,
+      "type": string,
+      "state": NavigationState.t,
+      "lastState": Js.Nullable.t(NavigationState.t),
+    } =>
+    unit
+  ) =>
+  {. "remove": unit => unit} =
```

Comment:
This can actually be `EventSubscription.t`, see https://github.com/reason-react-native/react-native/blob/master/src/apis/EventSubscription.re.